app/models/manageiq/providers/base_manager/event_catcher/runner.rb

```diff
@@ -110,6 +110,8 @@ def start_event_monitor
         monitor_events
       rescue EventCatcherHandledException
         Thread.exit
+      rescue ProviderUnreachable => err
+        raise ProviderUnreachable
```

Comment:
Calling `raise` without the class afterwards, will re-raise the original exception.  This way you don't lose the backtrace.